src/client/kotlin/tech/thatgravyboat/skyblockapi/api/profile/effects/EffectsAPI.kt

```diff
@@ -0,0 +1,120 @@
+import tech.thatgravyboat.skyblockapi.api.events.info.TabWidgetChangeEvent
+import tech.thatgravyboat.skyblockapi.api.events.misc.RegisterCommandsEvent
+import tech.thatgravyboat.skyblockapi.api.events.screen.ContainerInitializedEvent
+import tech.thatgravyboat.skyblockapi.api.profile.community.CommunityCenterAPI.cookieAteRegex
+import tech.thatgravyboat.skyblockapi.helpers.McClient
+import tech.thatgravyboat.skyblockapi.modules.Module
+import tech.thatgravyboat.skyblockapi.utils.extentions.cleanName
+import tech.thatgravyboat.skyblockapi.utils.extentions.getRawLore
+import tech.thatgravyboat.skyblockapi.utils.extentions.parseDuration
+import tech.thatgravyboat.skyblockapi.utils.extentions.parseWordDuration
+import tech.thatgravyboat.skyblockapi.utils.regex.RegexGroup
+import tech.thatgravyboat.skyblockapi.utils.regex.RegexUtils.anyMatch
+import tech.thatgravyboat.skyblockapi.utils.regex.RegexUtils.contains
+import tech.thatgravyboat.skyblockapi.utils.text.Text
+import tech.thatgravyboat.skyblockapi.utils.text.Text.send
+import tech.thatgravyboat.skyblockapi.utils.text.TextColor
+import tech.thatgravyboat.skyblockapi.utils.text.TextStyle.color
+import tech.thatgravyboat.skyblockapi.utils.time.until
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.days
+
+@Module
+object EffectsAPI {
+
+    val boosterCookieExpireTime get() = EffectsStorage.boosterCookieExpireTime
+
+
+    @Subscription
+    fun onChat(event: ChatReceivedEvent) {
+        if (cookieAteRegex.contains(event.text)) {
+            updateBoosterCookieExpireTime(boosterCookieExpireTime.until() + 4.days)
+        }
+    }
+
+    @Subscription
+    fun onInventoryFullyLoaded(event: ContainerInitializedEvent) {
+        if (event.title == "SkyBlock Menu") {
+            val cookieLore = event.itemStacks.find { it.cleanName == "Booster Cookie" }?.getRawLore() ?: return
+            cookieInventoryRegex.anyMatch(cookieLore, "duration") { (duration) ->
+                val parsedDuration = duration.parseDuration() ?: return@anyMatch
+                updateBoosterCookieExpireTime(parsedDuration)
+            }
+        }
+    }
+
+    @Subscription
+    fun onTabFooterUpdate(event: TabListHeaderFooterChangeEvent) {
+        val asshole = event.newFooterChunked.find { "Cookie Buff" in it } ?: return
+        asshole[2].let {
+            val parsedDuration = it.parseWordDuration() ?: return
+            updateBoosterCookieExpireTime(parsedDuration)
+        }
+    }
+
+    @Subscription
+    @OnlyWidget(TabWidget.ACTIVE_EFFECTS)
+    fun onTabWidgetUpdate(event: TabWidgetChangeEvent) {
+        cookieTabWidgetRegex.anyMatch(event.new, "duration") { (duration) ->
+            val parsedDuration = duration.parseDuration() ?: return@anyMatch
+            updateBoosterCookieExpireTime(parsedDuration)
+        }
+    }
+
+    private fun updateBoosterCookieExpireTime(parsedDuration: Duration) {
+        val expireTime = System.now() + parsedDuration
```

Comment:
boosterCookieExpireTime is an Instant so this should be parsedDuration.fromNow()